components/treelist/templates/column-chooser.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Column Chooser
+page_title: TreeList - Column Chooser Template
+description: Use the Column Chooser Template of the TreeList to control the rendering and customize the content of the Column Chooser.
+slug: treelist-templates-column-chooser
+tags: telerik,blazor,treelist,templates,column,menu
+published: True
+position: 40
+---
+
+# Column Chooser Template
+
+The Column Chooser Template lets you control the rendering of the [ColumnChooser]({%slug treelist-column-menu%}#column-chooser). It exposes a `context` object that contains a list of all columns in the TreeList. 
+
+When you set up the Column Chooser Template, the list of columns that are rendered by default will not be present.
+
+>caption Using the Template to provide custom rendering
+
+````CSHTML
+@* Use the Template to render the list of columns and add some custom styles. *@ 
+
+<TelerikTreeList Data="@TreeListData"
+                 IdField="@nameof(Employee.Id)"
+                 ParentIdField="@nameof(Employee.ParentId)"
+                 Pageable="true"
+                 Reorderable="true"
+                 Sortable="true"
+                 Resizable="true"
+                 FilterMode="@TreeListFilterMode.FilterMenu"
+                 ShowColumnMenu="true">
+                 <TreeListSettings>
+                     <TreeListColumnMenuSettings>
+                         <TreeListColumnMenuChooser>
+                             <Template>
+                                @{
+                                    var columns = context.Columns;
+                                    foreach(var column in columns)
+                                    {
+                                        <div style="border: solid 1px red">
+                                        <TreeListColumnMenuChooserItem Title="@column.DisplayTitle" ColumnId="@column.Id"></TreeListColumnMenuChooserItem>
+                                        </div>
+                                    }
+                                }
+                             </Template>
+                         </TreeListColumnMenuChooser>
+                     </TreeListColumnMenuSettings>
+                 </TreeListSettings>
+    <TreeListColumns>
+        <TreeListColumn Field="@nameof(Employee.Name)" Expandable="true" Width="320px" Id="name-column-id" />
+        <TreeListColumn Field="@nameof(Employee.Id)" Lockable="true" Id="id-column-id" />
+        <TreeListColumn Field="@nameof(Employee.ParentId)" Id="parentid-column-id" />
+        <TreeListColumn Field="@nameof(Employee.HireDate)" Id="hiredate-column-id" />
+        <TreeListColumn Field="@nameof(Employee.Salary)" Id="salary-column-id" />
+    </TreeListColumns>
+</TelerikTreeList>
+
+@code {
+    public List<Employee> TreeListData { get; set; }
+
+    async Task<List<Employee>> GetTreeListData()
```

Comment:
Follow our docs & demos code style guide (in the blazor repo wiki). In this case, all properties and methods must be `private`.